src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private static final Object[] DEFAULT_EMPTY_DATA_ARRAY = {};
```

Comment:
Don't use Objects, Arrays, or any other util class. You should implement the logic of increasing the array size yourself instead of using `Arrays.copyOf`.